### PR TITLE
Test stdarch (only x86 SSE and SSE2 for now)

### PR DESCRIPTION
**Pull request by eduardosm**
cc https://github.com/rust-lang/miri/pull/3088

### CODE REVIEW

File: run-stdarch-test.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        --skip test_mm_stream_ps
+        --skip test_mm_clflush
+        --skip test_mm_lfence
+        --skip test_mm_madd_epi16 # FIXME not yet implemented
+        --skip test_mm_maskmoveu_si128
+        --skip test_mm_mfence
+        --skip test_mm_stream_pd
+        --skip test_mm_stream_si128
+        --skip test_mm_stream_si32
+        --skip test_mm_stream_si64
```
Comment (RalfJung) [inline run-stdarch-test.sh:41]: Are these all nontemporal stores, or why don't they work in Miri?

**Comment by RalfJung** [comment]
This looks great, thanks a lot. :)

File: run-stdarch-test.sh (hunk 1 above)
Comment (eduardosm) [reply run-stdarch-test.sh:41]: They are non temporal stores, fences and CSR accesses

Comment (RalfJung) [reply run-stdarch-test.sh:41]: Okay. Some of the fences could maybe make sense, depends on what they fence, but for now this is fine I think.

File: run-stdarch-test.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+export TARGET="$1"
+
+case "$TARGET" in
+i586-*|i686-*|x86_64-*)
+    RUSTFLAGS="-C target-feature=+sse2"
+    TEST_ARGS=(
+        core_arch::x86::{sse,sse2}::
+        core_arch::x86_64::{sse,sse2}::
+        # FIXME add `#[cfg_attr(miri, ignore)]` to those tests in stdarch
```
Comment (RalfJung) [inline run-stdarch-test.sh:26]:
```suggestion
        # FIXME add `#[cfg_attr(miri, ignore)]` to those tests in stdarch
        # These are nontemporal stores, fences, and CSR (FP env status register) accesses
```

File: run-stdarch-test.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        --skip test_mm_comieq_ss_vs_ucomieq_ss
+        --skip test_mm_getcsr_setcsr_1
+        --skip test_mm_getcsr_setcsr_2
+        --skip test_mm_getcsr_setcsr_underflow
+        --skip test_mm_sfence
+        --skip test_mm_stream_ps
+        --skip test_mm_clflush
+        --skip test_mm_lfence
+        --skip test_mm_madd_epi16 # FIXME not yet implemented
```
Comment (RalfJung) [inline run-stdarch-test.sh:35]: This one looks like it should be outside the "FIXME add `#[cfg_attr(miri, ignore)]` to those tests in stdarch"?

**Pull request merged**